python/mujinplanningclient/realtimerobotplanningclient.py

```diff
@@ -1020,19 +1443,101 @@ def SetRobotBridgePause(self, timeout=10, **kwargs):
+            timeout (float, optional): Time in seconds after which the command is assumed to have failed. (Default: 1)
+            fireandforget (bool, optional): If True, does not wait for the command to finish and returns immediately. The command remains queued on the server.
+            jogtype (str, optional): **deprecated** One of 'joints', 'world', 'robot', 'tool'. Recommended to set this as a field in robotJogParameters instead.
+            checkSelfCollisionWhileJogging (bool, optional):
+            force (bool, optional): If true, forces the velocities to be set.
+            unit (str, optional): The unit of the given values. (Default: 'mm')
+            robotBridgeConnectionInfo (dict, optional): Information to set up a client to the robot bridge.
+            locationCollisionInfos (dict, optional): List of external collision IOs to be computed and sent in realtime.
+            robotJogParameters (dict, optional): A dictionary. Includes field 'jogtype' (One of 'joints', 'world', 'robot', 'tool').
+            simulationtimestep (float, optional): Time step of the simulation.
+            plotDirection (bool, optional): If True, plot the direction.
+        """
+        taskparameters = {
+            'command': 'SetJogModeVelocities',
+            'unit': unit,
+            'movejointsigns': movejointsigns,
+        }
+        if robotname is not None:
+            taskparameters['robotname'] = robotname
+        if toolname is not None:
+            taskparameters['toolname'] = toolname
+        if robotBridgeConnectionInfo is not None:
+            taskparameters['robotBridgeConnectionInfo'] = robotBridgeConnectionInfo
+        if locationCollisionInfos is not None:
+            taskparameters['locationCollisionInfos'] = locationCollisionInfos
+        if robotJogParameters is not None:
+            taskparameters['robotJogParameters'] = robotJogParameters
+        if simulationtimestep is not None:
+            taskparameters['simulationtimestep'] = simulationtimestep
+        if plotDirection is not None:
+            taskparameters['plotDirection'] = plotDirection
+        if robotspeed is not None:
+            taskparameters['robotspeed'] = robotspeed
+        if robotaccelmult is not None:
+            taskparameters['robotaccelmult'] = robotaccelmult
+        if canJogInCheckMode is not None:
+            taskparameters['canJogInCheckMode'] = canJogInCheckMode
+        if checkSelfCollisionWhileJogging is not None:
+            taskparameters['checkSelfCollisionWhileJogging'] = checkSelfCollisionWhileJogging
+        if force is not None:
+            taskparameters['force'] = force
+        taskparameters.update(kwargs)
+        return self.ExecuteCommand(taskparameters, timeout=timeout, fireandforget=fireandforget)
 
+    def EndJogMode(self, timeout=1, fireandforget=False, unit='mm', robotname=None, toolname=None, robotBridgeConnectionInfo=None, locationCollisionInfos=None, **kwargs):
```

Comment:
This function is not in master. It looks like the spec is outdated?

Reply:
Also SetJogModeVelocities. This definitely should not be part of this PR. Removed.

Overall tho - if a function is in the spec but not in master I always assume that we actually want to add it. Is it not the case? Its hard  to understand if EndJogMode is in the planning server or not. But the SetJogModeVelocities seems to be there.

Reply:
Master is the master. The spec must follow it except for when we added details to the spec (input/output format mainly). The client shouldn't have changed independently from the server.